### ISSUE TITLE
SASL PLAIN authentication failed: Kafka v0.10.1 SASL_SSL authentication not working with kafkajs versions after 2.2.0

### ISSUE THREAD

**Issue by AshishAYadav** [tulios/kafkajs#1483]
**Bug**

Kafka v0.10.1 SASL_SSL authentication not working with kafkajs versions after 2.2.0

SASL PLAIN authentication failed: The "chunk" argument must be of type string or an instance of Buffer or Uint8Array. Received an instance of ArrayBuffer"

Getting below error: 

{"level":"ERROR","timestamp":"2022-11-18T08:38:33.600Z","logger":"kafkajs","message":"[SaslAuthenticator-PLAIN] SASL PLAIN authentication failed: The \"chunk\" argument must be of type string or an instance of Buffer or Uint8Array. Received an instance of ArrayBuffer","broker":"ip-192-168-10-99:9092"}
{"level":"ERROR","timestamp":"2022-11-18T08:38:33.603Z","logger":"kafkajs","message":"[BrokerPool] SASL PLAIN authentication failed: The \"chunk\" argument must be of type string or an instance of Buffer or Uint8Array. Received an instance of ArrayBuffer","retryCount":0,"retryTime":339}
KafkaJSSASLAuthenticationError: SASL PLAIN authentication failed: The "chunk" argument must be of type string or an instance of Buffer or Uint8Array. Received an instance of ArrayBuffer
    at Object.authenticate (/home/ashish/aws-cli/node_modules/kafkajs/src/broker/saslAuthenticator/plain.js:18:23)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async SASLAuthenticator.authenticate (/home/ashish/aws-cli/node_modules/kafkajs/src/broker/saslAuthenticator/index.js:73:5)
    at async /home/ashish/aws-cli/node_modules/kafkajs/src/network/connection.js:139:9
    at async Connection.authenticate (/home/ashish/aws-cli/node_modules/kafkajs/src/network/connection.js:315:5)
    at async Broker.connect (/home/ashish/aws-cli/node_modules/kafkajs/src/broker/index.js:111:7)
    at async /home/ashish/aws-cli/node_modules/kafkajs/src/cluster/brokerPool.js:93:9
    at async /home/ashish/aws-cli/node_modules/kafkajs/src/cluster/index.js:107:14
    at async Cluster.connect (/home/ashish/aws-cli/node_modules/kafkajs/src/cluster/index.js:146:5)
    at async Object.connect (/home/ashish/aws-cli/node_modules/kafkajs/src/producer/index.js:219:7) {
  retriable: false,
  helpUrl: undefined,
  [cause]: undefined
}

**Code :**

```
import { Kafka, Partitioners } from 'kafkajs';
import * as fs from 'fs';

try {
    const Kafka_Client = new Kafka({
        clientId: "my-app",
        brokers: ["ip-192-168-10-99:9092"],
        // authenticationTimeout: 10000,
        // reauthenticationThreshold: 10000,
        ssl: {
            servername: 'ip-192-168-10-99',
            ca: [fs.readFileSync('./ssl/kafka/kafka-client-ca.pem', 'utf-8')],
            cert: fs.readFileSync('./ssl/kafka/kafka-client-cert-signed.pem', 'utf-8'),
            rejectUnauthorized: true,
        },
        sasl: {
            mechanism: "plain",
            username: "user",
            password: "secret",
        },
    });

    const producer = Kafka_Client.producer({ createPartitioner: Partitioners.LegacyPartitioner });

    await producer.connect();

    await producer.send({
        topic: "test",
        messages: [{ value: 'Hello KafkaJS user!' }],
    });
    console.log('success');
    await producer.disconnect();

} catch (error) {
    console.log(error);
}
```
**Expected behavior**
Connection should be successful with SASL_SSL protocol. 

**Environment:**
 - OS: [Ubuntu 20.04.5 LTS]
 - KafkaJS version [2.2.0, 2.2.1, 2.2.2]
 - Kafka version [0.10.1]
 - NodeJS version [v16.17.0, v14.15.0]

**Additional context**
However, it works fine with any version below kafkajs 2.2.0.
I cannot change kafka version in my production environment.

**Comment by ricardochen1996** [comment]
#1480

**Comment by Nevon** [comment]
Fixed in v2.2.3.

**Comment by AshishAYadav** [comment]
Thanks! @ricardochen1996 @Nevon
